### PR TITLE
exposed concealed ClassValue type in classnames package

### PR DESCRIPTION
**Pull request by jacamera**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Useful to be able to  `import { ClassValue } from 'classnames';` in order to reference the `ClassValue` type. Implemented according to the module template file for function modules.

### CODE REVIEW

**Comment by seansfkelley** [comment]
The typings are now part of the library itself; are you sure you want to make these changes?

https://github.com/JedWatson/classnames/blob/master/index.d.ts

File: types/classnames/index.d.ts [hunk 1]
```diff
@@ -9,21 +9,23 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-type ClassValue = string | number | ClassDictionary | ClassArray | undefined | null | boolean;
-
 interface ClassDictionary {
```
Comment (sheetalkamat) [inline types/classnames/index.d.ts]: please move all types except function into the namespace.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by adidahiya — CHANGES_REQUESTED** [review]
I'm pretty sure this declaration should be removed from this repo in favor of https://github.com/JedWatson/classnames/blob/master/index.d.ts

**Comment by jacamera** [comment]
@adidahiya The 2.2.6 version of classnames that is published on npm does not contain the TypeScript declaration file. I created an additional pull request on that repository to make the same changes.

https://github.com/JedWatson/classnames/pull/179

**Pull request closed without merging**